modules/pointcloudio/src/load_point_cloud.cpp

```diff
@@ -0,0 +1,96 @@
+#include "opencv2/load_point_cloud.hpp"
+
+#include "coders_base.hpp"
+#include "coders_obj.hpp"
+#include "coders_ply.hpp"
+
+#include <memory>
+
+
+namespace cv 
+{
+namespace pc
+{
+
+static PointCloudDecoder findDecoder( const String& filename ) 
+{
+    int filename_length = filename.length();
+    if (filename.substr(filename_length - 4, 4) == ".obj")
+    {
+        return std::unique_ptr<ObjDecoder>(new ObjDecoder());
+    }
+    if (filename.substr(filename_length - 4, 4) == ".ply")
+    {
+        return std::unique_ptr<PlyDecoder>(new PlyDecoder());
+    }
+
+}
+
+static PointCloudEncoder findEncoder( const String& filename ) 
+{
+    int filename_length = filename.length();
+    if (filename.substr(filename_length - 4, 4) == ".obj")
+    {
+        return std::unique_ptr<ObjEncoder>(new ObjEncoder());
+    }
+    if (filename.substr(filename_length - 4, 4) == ".ply")
+    {
+        return std::unique_ptr<PlyEncoder>(new PlyEncoder());
+    }
+
+}
+
+void loadPointCloud( const String& filename, OutputArray points, OutputArray normals )
+{
```

Comment:
OpenCV can be built without FS support for iOS and some other bare-metal configurations. The behavior is managed by `OPENCV_HAVE_FILESYSTEM_SUPPORT` macro. 
- See filesystem support in core: https://github.com/opencv/opencv/blob/master/modules/core/src/utils/filesystem.cpp
- Please add alternative branch with "not supported" exception like here: https://github.com/opencv/opencv/blob/master/modules/core/src/glob.cpp#L286. API call should not disappear from interface.